components/service/glean/src/test/java/mozilla/components/service/glean/storages/EventsStorageEngineTest.kt

```diff
@@ -350,4 +356,149 @@ class EventsStorageEngineTest {
             ) == snapshot.first().extra)
         assertEquals(1, testGetNumRecordedErrors(testEvent, ErrorType.InvalidValue))
     }
+
+    @Test
+    fun `continue recording events from disk`() {
+        val event = EventMetricType(
+            disabled = false,
+            category = "telemetry",
+            name = "test_event",
+            lifetime = Lifetime.Ping,
+            sendInPings = listOf("store1"),
+            allowedExtraKeys = listOf("foo")
+        )
+
+        // Record the event in the store, without providing optional arguments.
+        EventsStorageEngine.record(
+            metricData = event,
+            monotonicElapsedMs = SystemClock.elapsedRealtime(),
+            extra = mapOf("foo" to "bar")
+        )
+        EventsStorageEngine.testWaitForWrites()
+
+        assertEquals(1, EventsStorageEngine.eventStores["store1"]!!.size)
+        EventsStorageEngine.eventStores.clear()
+
+        // Reinitializing Glean should reload existing events on disk
+        Glean.initialized = false
+        FreshBootDetector.cacheIsFreshBoot = false
+        Glean.initialize(ApplicationProvider.getApplicationContext())
+
+        EventsStorageEngine.record(
+            metricData = event,
+            monotonicElapsedMs = SystemClock.elapsedRealtime()
+        )
+
+        assertEquals(2, EventsStorageEngine.eventStores["store1"]!!.size)
+
+        val snapshot = event.testGetValue()
+        assertEquals(2, snapshot.size)
+        assertEquals("telemetry", snapshot.first().category)
+        assertEquals("test_event", snapshot.first().name)
+        assertTrue(
+            "'extra' keys must be correctly recorded and truncated",
+            mapOf("foo" to "bar") == snapshot.first().extra
+        )
+    }
+
+    @Test
+    fun `flush events on fresh boot`() {
+        val server = MockWebServer()
+        server.enqueue(MockResponse().setBody("OK"))
+
+        resetGlean(getContextWithMockedInfo(), Glean.configuration.copy(
+            serverEndpoint = "http://" + server.hostName + ":" + server.port,
+            logPings = true
+        ))
+
+        val event = EventMetricType(
+            disabled = false,
+            category = "telemetry",
+            name = "test_event_clear",
+            lifetime = Lifetime.Ping,
+            sendInPings = listOf("store1"),
+            allowedExtraKeys = listOf("foo")
+        )
+
+        // Record the event in the store, without providing optional arguments.
+        EventsStorageEngine.record(
```

Comment:
Should we use `event.record()` here and use the testing API as well?

Reply:
Good idea.